examples/unified-format-function-calling/base_function_calling.py

```diff
@@ -0,0 +1,134 @@
+from pipecat.processors.aggregators.openai_llm_context import OpenAILLMContext
+from pipecat.services.ai_services import LLMService
+from pipecat.services.cartesia import CartesiaTTSService
+from pipecat.transports.services.daily import DailyParams, DailyTransport
+
+logger.remove(0)
+logger.add(sys.stderr, level="DEBUG")
+
+load_dotenv(override=True)
+
+
+async def start_fetch_weather(function_name, llm, context):
+    """Push a frame to the LLM; this is handy when the LLM response might take a while."""
+    await llm.push_frame(TTSSpeakFrame("Let me check on that."))
+    logger.debug(f"Starting fetch_weather_from_api with function_name: {function_name}")
+
+
+async def fetch_weather_from_api(function_name, tool_call_id, args, llm, context, result_callback):
+    await result_callback({"conditions": "nice", "temperature": "75"})
+
+
+class BaseFunctionCallingHandler:
+    """Generic base class for setting up and running an LLM-powered bot."""
+
+    def __init__(self, llm: LLMService):
+        """Initialize the base handler with a specific LLM."""
+        self.llm = llm
+
+    async def run(self):
+        """Set up and start the processing pipeline."""
+        async with aiohttp.ClientSession() as session:
+            (room_url, token) = await configure(session)
+
+            transport = DailyTransport(
+                room_url,
+                token,
+                "Respond bot",
+                DailyParams(
+                    audio_out_enabled=True,
+                    transcription_enabled=True,
+                    vad_enabled=True,
+                    vad_analyzer=SileroVADAnalyzer(),
+                ),
+            )
+
+            tts = CartesiaTTSService(
+                api_key=os.getenv("CARTESIA_API_KEY"),
+                voice_id="79a125e8-cd45-4c13-8a67-188112f4dd22",  # British Lady
+            )
+
+            # Register a function_name of None to get all functions
+            # sent to the same callback with an additional function_name parameter.
+            self.llm.register_function(
+                None, fetch_weather_from_api, start_callback=start_fetch_weather
+            )
+
+            weather_function = FunctionSchema(
+                name="get_current_weather",
+                description="Get the current weather",
+                properties={
+                    "location": {
+                        "type": "string",
+                        "description": "The city and state, e.g. San Francisco, CA",
+                    },
+                    "format": {
+                        "type": "string",
+                        "enum": ["celsius", "fahrenheit"],
+                        "description": "The temperature unit to use. Infer this from the user's location.",
+                    },
+                },
+                required=["location"],
+            )
+            tools = ToolsSchema(standard_tools=[weather_function])
```

Comment:
Why in this example this doesn't go into a `static_method`?

Reply:
Hi @aconchillo, I'm not sure I understood the question. Do you mean transforming everything into a static method instead of having the class `BaseFunctionCallingHandler` ?